File: 1911.最大子序列交替和.c
/*
 * @lc app=leetcode.cn id=1911 lang=c
 *
 * [1911] 最大子序列交替和
 */

// @lc code=start
long long maxAlternatingSum(int* nums, int numsSize) {
    long long even = nums[0], odd = 0;
    for (int i = 0; i < numsSize; i++) {
        even = fmax(even, odd + nums[i]);
        odd = fmax(odd, even - nums[i]);
    }
    return even;
}
// @lc code=end

File: 1911.最大子序列交替和.go
package leetcode

/*
 * @lc app=leetcode.cn id=1911 lang=golang
 *
 * [1911] 最大子序列交替和
 */

// @lc code=start
func maxAlternatingSum(nums []int) int64 {
	even, odd := nums[0], 0
	for i := 0; i < len(nums); i++ {
		even = max(even, odd+nums[i])
		odd = max(odd, even-nums[i])
	}
	return int64(even)

}

// @lc code=end
